curvlinops/kfac.py

```diff
@@ -365,20 +394,12 @@ def draw_label(self, output: Tensor) -> Tensor:
             return output.clone().detach() + perturbation
 
         elif isinstance(self._loss_func, CrossEntropyLoss):
-            # TODO For output.ndim > 2, the scale of the 'would-be' gradient resulting
-            # from these labels might be off
-            if output.ndim != 2:
```

Comment:
Do you think it makes sense to add a safeguard at the beginning of this function that `assert`s `output.ndim==2`?

I like the idea of collecting all axes that are treated like batch axes in a single dimension. Just want to make sure that we remember we're using this idea whenever we use `draw_label` in the future (potentially for other curvature approximations).

Reply:
Also, making this assumption within `draw_label` simplifies the code for `MSELoss`. If we decide to make this assumption, can you refactor the code handling `MSELoss`, too?

Reply:
Not sure in what sense the code for `MSELoss` can be more simplified (besides changing the `std` for the `"mean"` case; see my latest commit).

Reply:
I meant that, but when I wrote it down I assumed the diff would be bigger 😂. Thanks for fixing this!